### DISCUSSION TITLE
Using Pymor with two-mass oscillator

### DISCUSSION THREAD

**Discussion by NoraYV** [pymor/pymor#1597] [Q&A] [answered]
I want to use pyMor for a two-mass oscillator. As you see in the image it is a single-input-single-output system.
<img width="394" alt="ZMS" src="https://user-images.githubusercontent.com/100579311/158171633-9175d0f2-d866-47d7-b71d-9666de8002cd.png">
I followed the pyMor truncated balancing tutorial. As soon as I got to create the bode plots I got confused. Like in the tutorial I want to create a plot included the fom an rom system.  I use:

```python
fig, axs = plt.subplots(2, 1, figsize=(6, 12), sharex=True, constrained_layout=True)
fom.transfer_function.bode_plot(w, ax=axs)
_ = rom.transfer_function.bode_plot(w, ax=axs, linestyle='--')
plt.show()
```

but it doesn't work. I get the Error:

```pytb
Traceback (most recent call last):
  File "C:\Users\Yazdandoost\PycharmProjects\Loeschen\2-Massen-Schwinger.py", line 47, in <module>
    fom.transfer_function.bode_plot(w, ax=axs)
  File "C:\Users\Yazdandoost\Anaconda3\envs\Loeschen\lib\site-packages\pymor\models\transfer_function.py", line 206, in bode_plot
    assert isinstance(ax, np.ndarray) and ax.shape == (2 * self.dim_output, self.dim_input)
AssertionError
```

**What is the reason that this code work in the tutorial but not in my example?**

**My complete code is listed here:**

```python
from functools import reduce
import matplotlib.pyplot as plt
import numpy as np
import scipy.sparse as sps
from pymor.models.iosys import LTIModel
from pymor.reductors.bt import BTReductor

plt.rcParams['axes.grid'] = True

k = 4
E = sps.eye(k, format='lil') # Einheitsmatrix
E = E.tocsc()
A= np.array([[0,0,1,0], [0,0,0,1], [-5, 3, 0, 0], [3, -3, 0, 0]])
B = np.array([[0], [0], [0], [1]])
C = np.array([[1,0,0,0]])

fom = LTIModel.from_matrices(A, B, C, E=E)
bt = BTReductor(fom)

error_bounds = bt.error_bounds()
print('error bounds:', '\n', error_bounds)
hsv = fom.hsv()
fig, ax = plt.subplots()
ax.semilogy(range(1, len(error_bounds)+1), error_bounds, '.-')
ax.semilogy(range(1, len(hsv)), hsv[1:], '.-')
ax.legend(['error_bounds', 'hsv'])
ax.set_xlabel('Reduced order')
_ = ax.set_title(r'Upper and lower $\mathcal{H}_\infty$ error bounds')
plt.show()

rom = bt.reduce(2)

w = np.logspace(-1, 50, 300)

fig, axs = plt.subplots(2, 1, figsize=(6, 12), sharex=True, constrained_layout=True)
fom.transfer_function.bode_plot(w, ax=axs)
_ = rom.transfer_function.bode_plot(w, ax=axs, linestyle='--')
plt.show()
```

**Comment by pmli** [comment, marked as answer]
The issue seems to be in `plt.subplots`. By default, it will squeeze the shape of the `axs` array such that `axs.shape == (2,)` instead of `axs.shape == (2, 1)`. To prevent this, add `squeeze=False` to `plt.subplots(...)`.

(BTW, I updated your post to show syntax highlighting for code blocks. There is some information [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks).)

**Reply by NoraYV** [reply]
Thank you for this answer and the information to write the code correct!
